pkg/controller/buildrun/buildrun_controller.go

```diff
@@ -116,6 +116,18 @@ func (r *ReconcileBuildRun) Reconcile(request reconcile.Request) (reconcile.Resu
 		return reconcile.Result{}, fmt.Errorf("errors: %v %v", err, updateErr)
 	}
 
+	if buildRun.GetLabels() == nil {
+		buildRun.Labels = make(map[string]string)
+	}
+	if buildRun.GetLabels()[buildv1alpha1.LabelBuild] == "" {
```

Comment:
@zhangtbj wondering why we do this. So if I get this right, we just set labels in the BuildRun, where we already have an snapshot of the Build. What is the aggregated value of doing this? Asking because this ends-up with a client update call, which if possible we should avoid.

Reply:
Set the label is for query all buildrun which under one build easily:
```
	lbls := map[string]string{
		buildv1alpha1.LabelBuild: b.Name,
	}
	opts := client.ListOptions{
		Namespace:     b.Namespace,
		LabelSelector: labels.SelectorFromSet(lbls),
	}
	if err := r.client.List(context.TODO(), buildRunList, &opts); err != nil {
		return err
	}
```

The snapshot is just for this buildrun
The label on the buildrun is easy for query all buildruns.
It is good for clean up and future query.

Reply:
What are your thoughts on adding "ownerReferences" to BuildRuns? I guess in this way we could avoid adding labels to objects with the only purpose of filtering based on labels prior to deletion

Reply:
Yep, Enrique.

I think about that, but I think it is not good for this case because you don't know all the buildrun ownerReferences status in each reconciliation. So every time, you need to go through ALL buildruns which under this build to add more remove the `ownerReferences`

It is not convenient than this way and the performance is not good thant just delete in the end.

Reply:
@zhangtbj i´m not sure, I need to do a comparison first. I understood that with ownerReferences is simple. I will make myself a todo for this, and maybe PR something if my investigations probe that ownerreference is better.